lib/auth.js

```diff
@@ -0,0 +1,148 @@
+/* global XMLHttpRequest */
+
+var defaultConfig = require('./config-default')
+
+/**
+ * Returns whether or not the current JS engine is in a browser environment. The
+ * environment is assumed to be Node.js if this returns false. Heuristically
+ * just looks for 'window' and 'XMLHttpRequest'.
+ * @returns {Boolean} true if the environment is a browser, false otherwise
+ */
+function isBrowser () {
+  return typeof window !== 'undefined' && typeof XMLHttpRequest !== 'undefined'
+}
+
+/**
+ * Enumerates the two main types of authentication endpoints.
+ * @readonly
+ * @enum {number}
+ */
+const AUTH_ENDPOINTS = {
+  PRIMARY: 0,
+  SECONDARY: 1
+}
+
+/**
+ * "Logs in"* to a WebID-TLS server by sending a HEAD request.  The TLS
+ * handshake happens implicitly through either an 'XMLHttpRequest' or node
+ * 'https.request'.
+ *
+ * First tries to log in to the "primary" WebID-TLS endpoint, and retries with
+ * the "fallback" endpoint if the first attempt fails.
+ *
+ * @param {Object} config - the configuration object specified in {@link
+ * module:config-default}.
+ * @returns {(Promise.<String>|Promise.<null>)} the WebID as a string if the
+ * client cert is recognized, otherwise null.
+ */
+function login (config) {
+  config = Object.assign({}, defaultConfig, config)
+  return loginTo(AUTH_ENDPOINTS.PRIMARY, config)
+    .then(webId => webId || loginTo(AUTH_ENDPOINTS.SECONDARY, config))
+}
+
+/**
+ * Alias to {@link login}
+ */
+const currentUser = login
+
+/**
+ * Logs in to the specified endpoint with the given configuration
+ * @param {AUTH_ENDPOINTS} endpoint - the endpoint type
+ * @param {@link module:config-default} config - the config object
+ * @returns {(Promise.<String>|Promise.<null>)} the WebID as a string if the
```

Comment:
You can just do `Promise<string>` -- the `Promise.<>` notation is being deprecated. (Plus it looks better without the .)